### DISCUSSION TITLE
Implement NIS code library

### DISCUSSION THREAD

**Discussion by paoloros97** [omersiar/esp-rfid#494] [Ideas]
A nice idea is to **implement** also the **library to read the NIS code** inside the CIE (Carta d'Identità Elettronica italiana) from [Developers Italia](https://github.com/italia).
This is the repository: https://github.com/italia/cie-PN532
Info about NIS code: https://docs.italia.it/italia/cie/cie-accessi-pinless-manuale-docs/it/stabile/introduzione.html

I believe it can be a safer solution to the UID code since it can be copied easily to another changeable UID card.

Problem: at the moment it's implemented to work only with the PN532 RFID reader.

**Comment by matjack1** [comment]
Hey @paoloros97 that's a very nice idea! We support already the PN532, so it should be possible in theory :)

I don't know how the CIE works, but why do you think it's more difficult to clone?

I think that the only improved approach that we can do right now is adding a personal pin code, so that you swipe the card and then enter a pin code. Still quite easy to hack with a replay attack, but butter than a simple swipe with a cloned card :)

**Reply by paoloros97** [reply]
I've already asked info about this on Forum Italia: https://forum.italia.it/t/clone-cie/18720
It also slightly help the fact that it's illegal (or criminally prosecutable) to make clones of CIE.
